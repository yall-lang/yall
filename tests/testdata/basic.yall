{hey}

(hello "there")

(map
	(hello (friend "bread"))
	(hello (friend "lilac")))

[1 2 3]
