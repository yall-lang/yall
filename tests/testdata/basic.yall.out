[
    Expression {
        kind: Block,
        values: [
            Identifier(
                "hey",
            ),
        ],
    },
    Expression {
        kind: Item,
        values: [
            Identifier(
                "hello",
            ),
            Text(
                "there",
            ),
        ],
    },
    Expression {
        kind: Item,
        values: [
            Identifier(
                "map",
            ),
            Expression(
                Expression {
                    kind: Item,
                    values: [
                        Identifier(
                            "hello",
                        ),
                        Expression(
                            Expression {
                                kind: Item,
                                values: [
                                    Identifier(
                                        "friend",
                                    ),
                                    Text(
                                        "bread",
                                    ),
                                ],
                            },
                        ),
                    ],
                },
            ),
            Expression(
                Expression {
                    kind: Item,
                    values: [
                        Identifier(
                            "hello",
                        ),
                        Expression(
                            Expression {
                                kind: Item,
                                values: [
                                    Identifier(
                                        "friend",
                                    ),
                                    Text(
                                        "lilac",
                                    ),
                                ],
                            },
                        ),
                    ],
                },
            ),
        ],
    },
    Expression {
        kind: List,
        values: [
            Number(
                "1",
            ),
            Number(
                "2",
            ),
            Number(
                "3",
            ),
        ],
    },
]
